### DISCUSSION TITLE
One Entitas version - Desperate Devs is open-source now

### DISCUSSION THREAD

**Discussion by sschmid** [sschmid/Entitas-CSharp#1008] [Announcements]
# Hi Entitas community!

### Great news: Desperate Devs is now open-source:
https://github.com/sschmid/DesperateDevs

---

### Other great news: There is only one Entitas version that has it all!

Entitas was, is and will be open-source.

However, there have been two different versions:
- [Entitas GitHub version](https://github.com/sschmid/Entitas/releases)
- [Entitas Unity Asset Store version](http://u3d.as/NuJ)

The Asset Store version contained `Jenny.zip`, which included additional code generators that have not been part of the GitHub version.
The Asset Store version also came with a specific code generator that was based on [Roslyn](https://github.com/dotnet/roslyn) instead of reflection, which enabled us to generate code, even if there are compiler errors.

This changed now as I made Desperate Devs (including Jenny and the Roslyn code generators) open-source! The Entitas GitHub version is now the default version and contains everything that the Asset Store version had.

---

### Entitas for Unity

The [Entitas releases](https://github.com/sschmid/Entitas/releases) now contain both `Entitas.zip` and `Jenny.zip`. Drop the content of `Entitas.zip` into your Assets folder and the content of `Jenny.zip` into the root of your project folder.

See the Unity project [Match-one](https://github.com/sschmid/Match-One) for reference.

---

### Entitas for C# projects

Entitas is also available via [nuget](https://www.nuget.org), see the dotnet console app [Entitas-Standalone](https://github.com/sschmid/Entitas-Standalone) for reference.

---

Happy coding :)
Simon

**Comment by FNGgames** [comment]
This is amazing, thank you Simon.

**Comment by Jennal** [comment]
Thank you for your great job, Simon. Are you planning to make Jenny open source soon?

**Reply by sschmid** [reply]
It is already, see [Desperate Devs repo](https://github.com/sschmid/DesperateDevs)
